cmd/natc/natc.go

```diff
@@ -244,6 +309,21 @@ var tsMBox = dnsmessage.MustNewName("support.tailscale.com.")
 // argument is the NodeID of the node that sent the request.
 func (c *connector) generateDNSResponse(req *dnsmessage.Message, from tailcfg.NodeID) ([]byte, error) {
 	pm, _ := c.perPeerMap.LoadOrStore(from, &perPeerState{c: c})
+	var addrs []netip.Addr
+	if len(req.Questions) > 0 {
+		switch req.Questions[0].Type {
+		case dnsmessage.TypeAAAA, dnsmessage.TypeA:
+			var err error
+			addrs, err = pm.ipForDomain(req.Questions[0].Name.String())
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+	return dnsResponse(req, addrs)
+}
+
+func dnsResponse(req *dnsmessage.Message, addrs []netip.Addr) ([]byte, error) {
```

Comment:
docs

Reply:
and a better name, `generateDNSResponse` fits but its also used by `c`.

Reply:
I don't want to 'Resolve conversation' in github without asking: do you have any thoughts for the better name? `dnsResponse` was my favorite I could come up with.

Reply:
this is fine now.